samcli/lib/build/app_builder.py

```diff
@@ -216,20 +216,26 @@ def update_template(template_dict: Dict, original_template_path: str, built_arti
 
             resource_type = resource.get("Type")
             properties = resource.setdefault("Properties", {})
+            package_type = properties.get("PackageType", ZIP)
 
-            if resource_type == SamBaseProvider.SERVERLESS_FUNCTION and properties.get("PackageType", ZIP) == ZIP:
-                properties["CodeUri"] = store_path
+            if resource_type == SamBaseProvider.SERVERLESS_FUNCTION and package_type == ZIP:
```

Comment:
You should do it like in other functions:
```python
if package_type == ZIP:
    if resource_type == SamBaseProvider.SERVERLESS_FUNCTION and ("InlineCode" not in properties or properties["InlineCode"] is None):
        properties["CodeUri"] = store_path
    elif resource_type == SamBaseProvider.LAMBDA_FUNCTION and ("Code" not in properties or "ZipFile" not in properties["Code"] or properties["Code"]["ZipFile"] is None):
        properties["Code"] = store_path
elif package_type == IMAGE:
    if resource_type == SamBaseProvider.LAMBDA_FUNCTION:
        properties["Code"] = built_artifacts[logical_id]
    elif resource_type == SamBaseProvider.SERVERLESS_FUNCTION:
        properties["ImageUri"] = built_artifacts[logical_id]
```

Reply:
will probably not refactor it this way because we have another resource type check for layers, would probably make all conditions look parallel!